### PR TITLE
[AppService] Support list, delete, approve and reject private endpoint connection for webapp in CLI

### LINKED ISSUE #13072: webapp: support connecting privately to a Web App using Azure Private Endpoint
**Is your feature request related to a problem? Please describe.**
https://docs.microsoft.com/en-us/azure/private-link/create-private-endpoint-webapp-portal

**Describe the solution you'd like**

**Update:** 

Changed to support connecting privately to a Web App using private endpoint connections under `az network private-endpoint-connection`

The following commands:
`az network private-link-service list`,
`az network private-endpoint-connection (approve, delete, list, reject, show)`
Should work with `--type Microsoft.Web/sites`

**Before:**

Command | CLI | PowerShell | Description
-- | -- | -- | --
Get Private Endpoint Connection. Takes: site info + private endpoint connection name. Returns: private endpoint connection info | Az webapp private-endpoint-connection list | Get-AzWebAppPrivateEndpointConnection | Gets the details.  If the optional private endpoint name is specified, will return the details for just that one private endpoint.  If just the app name is specified, will return the details for all of the private endpoints on that app. Arguments: sitename, optional:private endpoint name
Delete Private Endpoint Connection. Takes: site info + private endpoint connection name | Az webapp private-endpoint-connection delete | Delete-AzWebAppPrivateEndpointConnection | Removes a private endpoint from an app. Arguments: sitename private endpoint name
Approve Private Endpoint Connection. Takes: site info + private endpoint connection name. Returns: private endpoint connection info | Az webapp private-endpoint-connection approve | Set-AzWebAppPrivateEndpointConnection | Sets the approve status on  requested private endpoint connection with an app. CLI Arguments: sitename private endpoint name. Powershell Arguments: sitename private endpoint name  "approve"
Reject Private Endpoint Connection. Takes: site info + private endpoint connection name. Returns: private endpoint connection info | Az webapp private-endpoint-connection reject | Set-AzWebAppPrivateEndpointConnection | Sets the reject status on  requested private endpoint connection with an app. CLI Arguments: sitename private endpoint name. Powershell Arguments: sitename private endpoint name  "reject"

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by calvinsID**
**Description**  

Closes #13072 

Add support for Webapp in the network private-endpoint-connection & private-link-resource commands

**Testing Guide**  

The existing commands:
`az network private-link-resource list`
`az network private-endpoint-connection (approve, delete, list, reject, show)`
Should now work with `--type Microsoft.Web/sites`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by Juliehzl** [comment]
@calvinsID  are you working on private endpoint connection?

**Comment by Juliehzl** [comment]
@MyronFanQiu Can we utilize the rest method for webapp?

**Comment by MyronFanQiu** [comment]
@calvinsID https://github.com/Azure/azure-cli/pull/13608 Please follow this PR to support private link and private endpoint scenario in Azure CLI. My alias is fanqiu. If you have any problem, feel free to ping me. I'm changing our doc right now.

**Comment by calvinsID** [comment]
@Juliehzl I should clarify, just working on the list, delete, approve & reject commands as outlined in https://github.com/Azure/azure-cli/issues/13072

**Comment by yonzhan** [comment]
AppService Private Link

File: src/azure-cli/azure/cli/command_modules/appservice/commands.py [hunk 1]
```diff
@@ -216,6 +216,12 @@ def load_command_table(self, _):
         g.custom_command('add', 'add_hc')
         g.custom_command('remove', 'remove_hc')
 
+    with self.command_group('webapp private-endpoint-connection', is_preview=True) as g:
```
Comment (MyronFanQiu) [inline src/azure-cli/azure/cli/command_modules/appservice/commands.py:219]: talking with Network team, these commands should be in `az network` command group. Could you follow this [doc](https://github.com/Azure/azure-cli/blob/dev/doc/private_endpoint_connection_command_guideline.md)

**Pull request merged**